service/collector.go

```diff
@@ -246,7 +201,11 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	return nil
 }
 
-func (col *Collector) execute(ctx context.Context) error {
+// Run starts the collector according to the command and configuration
+// given by the user, and waits for it to complete.
+// Consecutive calls to Run are not allowed, Run shouldn't be called
+// once a collector is shut down.
+func (col *Collector) Run(ctx context.Context) error {
```

Comment:
Should we add a unit test calling `Run` without the command wrapper?

Reply:
Ah, the `TestCollector_StartAsGoRoutine` already does it. No need to add a new one then.

Reply:
I've just changed that to do it based on your feedback 😄

Reply:
hard to see with the force push haha, LGTM then :)